### DISCUSSION TITLE
GitHubCopilit open source alternatives

### DISCUSSION THREAD

**Discussion by theCollectiv** [mason-org/mason.nvim#1360] [General]
I'm not sure, if it fits* in right here, but maybe it's worth a thought.
There are 2 'GitHubCopilit open source alternatives' around at the moment:
- tabbyML [TabbyML/tabby: Self-hosted AI coding assistant](https://github.com/TabbyML/tabby)
- [fauxpilot/fauxpilot: FauxPilot - an open-source alternative to GitHub Copilot server](https://github.com/fauxpilot/fauxpilot)

Both are
- in a relative early stage of development (but very promising) 
- ran in a docker container to provide the server on the local machine

Maybe it could be a great idea to provide them via Mason ;-)

* it doesn't fall in the categories of an LSP, DAP, Linter, Formatter ...but it's like a nvim as client for a server service (I guess this kind of counts).

**Comment by williamboman** [comment]
As long as they provide a standalone binary I think these would be a good fit for Mason. They'd not fit to any existing category, but that's fine. From a very quick investigation, it only seems Tabby fits this requirement (and they only provide an aarch64 Mac binary).
